parts/linux/cloud-init/artifacts/components.json

```diff
@@ -8,512 +10,18 @@
         "1.8.22"
       ]
     },
+
+    //new
     {
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/metrics-server:*",
+      "name": "nvidia-cuda",
+      "type": "containerimages //extend to OCI |Tar|tgz when merging with packages,
+      "action": "cacheandfetch|fetchonly|config",  //where cacheandfetch is cached at build time and also to fetch at provisioning time if not cached. fetchonly is only fetching at provisioning time but not caching. config is neither cache nor fetch but just for storing data for other purpose. Open to change the names of the values.
```

Comment:
for action, I'd probably use `cache`/`fetch`/`config` (also, what will config actually be used for?), where
`cache` signifies that we must always have the component `cache`d  by the time provisioning ends (whether that be during build or provisioning time doesn't matter), and `fetch`signifies we only fetch at runtime

Reply:
runtime != provisioning time? What is the difference?

I thought there are only 2 timing to download/install a package? One is at building in the packer VM and the other is RP passing CSE when (provisioning) bootstrapping a node?

Reply:
runtime == provisioning time, I use the terms interchangeably which is probably something I should stop doing

my suggestion is:

`cache` = build time/provisioning time
`fetch` = provisioning time only

Reply:
I see. That makes sense.
One thing comes to me: Recently `cni` removed fetching at provisioning time. Instead it just uses whatever cached at build time. So in this case, it's the feature owner who determines how to react to the `cache`, `fetch` value from components.json. For example, in components.json, `cni` set to `cache`. But in the scripts, it will only cache it but not also fetch it.